File: frontend/next-generation-ticketing-system/src/Entity/CustomEntity/TicketView.ts
import Ticket from "../Ticket";
import BusSchedule from "../BusSchedule";
import TicketAddress from "../TicketAddress";
import Seat from "../Seat";
import RouteDistrict from "../RouteDistrict";

class TicketView {
  ticket: Ticket;
  busSchedule: BusSchedule;
  ticketAddress: TicketAddress;
  seats: Seat[];
  routeDistricts: RouteDistrict[];

  constructor(
    ticket: Ticket,
    busSchedule: BusSchedule,
    ticketAddress: TicketAddress,
    seats: Seat[],
    routeDistricts: RouteDistrict[]
  ) {
    this.ticket = ticket;
    this.busSchedule = busSchedule;
    this.ticketAddress = ticketAddress;
    this.seats = seats;
    this.routeDistricts = routeDistricts;
  }
}

export default TicketView;

File: backend/ngts/src/main/java/com/three/ngts/Entity/CustomEntity/TicketView.java
package com.three.ngts.Entity.CustomEntity;

import com.three.ngts.Entity.*;
import lombok.Data;

import java.util.List;

@Data
public class TicketView {
    private Ticket ticket;
    private BusSchedule busSchedule;
    private TicketAddress ticketAddress;
    private List<Seat> seats;
    private List<RouteDistrict> routeDistricts;

    public TicketView(Ticket ticket, BusSchedule busSchedule, TicketAddress ticketAddress, List<Seat> seats,
            List<RouteDistrict> routeDistricts) {
        this.ticket = ticket;
        this.busSchedule = busSchedule;
        this.ticketAddress = ticketAddress;
        this.seats = seats;
        this.routeDistricts = routeDistricts;
    }
}
